providers/implementations/keymgmt/mlx_kmgmt.c

```diff
@@ -710,9 +764,19 @@ static void *mlx_kem_gen(void *vgctx, OSSL_CALLBACK *osslcb, void *cbarg)
     if ((gctx->selection & OSSL_KEYMGMT_SELECT_KEYPAIR) == 0)
         return key;
 
-    /* For now, using the same "propq" for all components */
-    key->mkey = EVP_PKEY_Q_keygen(key->libctx, key->propq,
-                                  key->minfo->algorithm_name);
+    adjusted_propq = get_adjusted_propq(propq);
+    /* For pre-3.2 FIPS providers we adjust "propq" for all PQ part and pass it downwards */
```

Comment:
If we want to make a pre-3.2 determination then we actually should be checking that - as it is this is entirely unconditional code that impacts all versions ...